ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -743,8 +745,15 @@ let on_notification server (notification : Client_notification.t) :
   | WillSaveTextDocument _
   | Initialized
   | WorkDoneProgressCancel _
-  | WorkDoneProgress _
-  | Exit -> Fiber.return state
+  | WorkDoneProgress _ -> Fiber.return state
+  | Exit ->
+    let exit_code =
+      (* https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#exit *)
+      match state.shutdown_status with
+      | Shutdown_received -> 0
+      | Shutdown_not_received -> 1
+    in
+    exit exit_code
```

Comment:
This looks wrong. With this, any sub-processes launched by lsp would not be cleaned up

Reply:
> This looks wrong. With this, any sub-processes launched by lsp would not be cleaned up

How does `Client.stop` handle this? I can see some finalization steps being taken here

https://github.com/ocaml/ocaml-lsp/blob/cb338c754bcc99c329fb45e0c4554c387cd185f2/ocaml-lsp-server/src/ocaml_lsp_server.ml#L828-L837

but I have no idea if they apply for the sub-processes being referred to.